### ISSUE TITLE
Emissary can't find tlsSecret for Ingress resource

### ISSUE THREAD

**Issue by nuclear0wl** [emissary-ingress/emissary#3959]
**Describe the bug**
Hello, when I'm trying to create Ingress resource with TLS enables using the manifest below:
```
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  name: kubia-ambassador-ingress
  annotations:
    kubernetes.io/ingress.class: ambassador
spec:
  tls:
  - hosts:
      - host.example.com
    secretName: example-tls-secret
  rules:
  - host: host.example.com
    http:
      paths:
      - path: /kubia
        pathType: Prefix
        backend:
          service:
            name: kubia-service
            port:
              number: 80
```
I'm getting the following error:
```
2021-12-02 16:50:31 diagd 2.0.5 [P21TAEW] ERROR: Secret example-tls-secret.default unknown
2021-12-02 16:50:31 diagd 2.0.5 [P21TAEW] ERROR: Host kubia-ambassador-ingress-0-0: invalid TLS secret example-tls-secret, marking inactive
```
At the same time if I will use native emissary resources then all is Ok with the same secret:
```
apiVersion: getambassador.io/v3alpha1
kind: Host
metadata:
  name: minimal-host
spec:
  tlsSecret:
    name: example-tls-secret
  hostname: host.example.com
  mappingSelector:
    matchLabels:
      examplehost: host
  requestPolicy:
    insecure:
      action: Route
---
apiVersion: getambassador.io/v3alpha1
kind: Mapping
metadata:
  name: quote-backend
spec:
  hostname: "*"
  prefix: /backend/
  service: kubia-service:80
```
Also I've tried to obtain TLS certificate automatically using **cert-manager** and in such case I'm getting the absolutely same error.

I have a bunch of my test resources in namespace _default_ and emissary-ingress in a separate one. Do I miss something or may this be a bug? Can you please help to understand.

**To Reproduce**
Deploy Ingress resource with TLS secret presented.

**Expected behavior**
Ingress resource works absolutely the same way as Host + Mapping resources.

**Versions (please complete the following information):**
 - Ambassador: 2.0.5
 - Azure Kubernetes Service
 - Version v1.19.11
 - Helm Chart version emissary-ingress-7.1.10

**Comment by volatilemolotov** [comment]
This is still an issue in Emissary 2.2.2. workaround is to create the following TLSContext and referencing the Host that was auto-created for Ingress resource. Such host does not exist in the K8S api but Emissary logs show that it is named $ingressname-0-0 so the TLSContext looks like this :

```
apiVersion: getambassador.io/v3alpha1
kind: TLSContext
metadata:
  name: ingress-tls
spec:
  hosts: 
    - ingressname-0-0
  secret: ingress-tls-prod
```

**Comment by cindymullins-dw** [comment]
Closing to prioritize issues without available workarounds.

**Comment by eevdev** [comment]
The fix that my colleague added in https://github.com/emissary-ingress/emissary/pull/4607 works for my case at least. Hoping it'll be merged quickly, but can be tested with this image in the meantime: olemarkus/emissary:3.2.1-0.20221006115358-fdcb80b50ad7-dirty.1665227182

**Comment by AliceProxy** [comment]
Let's not just close issues because they have a workaround. Closing means we're not going to do anything about the issue. Not the same as de-prioritizing it over other items.

**Comment by AliceProxy** [comment]
Fixed via https://github.com/emissary-ingress/emissary/pull/4610 thanks to @olemarkus. Should be available in the next release of Emissary/Edge Stack (sometime this week or early next week). I'll close out this issue after the release.

**Comment by AliceProxy** [comment]
Ok, I'm going to close this issue now since we [just released emissary 3.3.0](https://github.com/emissary-ingress/emissary/releases/tag/v3.3.0) with the fix for this. Please re-open this issue if you have any problems with the fix in that release.
